test/integration/targets/loops/tasks/main.yml

```diff
@@ -176,3 +176,29 @@
   with_sequence: start=0 count=3
   loop_control:
     index_var: my_idx
+
+#
+# loop_control/label
+# https://github.com/ansible/ansible/pull/36430
+#
+
+- set_fact:
+    loopthis:
+        - name: foo
+          label: foo_label
+        - name: bar
+          label: bar_label
+
+- name: check that item label is updated each iteration
+  debug:
+    msg: "{{ looped_var.name }}"
+  with_items: "{{ loopthis }}"
+  loop_control:
+    loop_var: looped_var
+    label: "looped_var {{ looped_var.label }}"
+  register: output
+
+- assert:
+    that:
+      - "output.results[0]['_ansible_item_label'] == 'looped_var foo_label'"
+      - "output.results[1]['_ansible_item_label'] == 'looped_var bar_label'"
```

Comment:
_ansible_item_label is gauranteed to always be in the looped results?

Reply:
yes, even if user does not define it, you get the 'default label value'  `{{ item }}`